core/src/test/java/org/apache/james/core/healthcheck/ResultStatusTest.java

```diff
@@ -0,0 +1,64 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core.healthcheck;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+public class ResultStatusTest {
+
+    @Test
+    public void mergeWithReturnHealthyWhenMergeWithNoResult() {
+        ResultStatus healthy = ResultStatus.HEALTHY;
+
+        assertThat(healthy.mergeWith(null)).isEqualTo(ResultStatus.HEALTHY);
+    }
+
+    @Test
+    public void mergeWithReturnUnHealthyWhenMergeWithUnHealthy() {
```

Comment:
We tend to only do one assert per test.

Here specifically, the issue is that if one test is failing, the test will be aborted. Somùe other assertions might be failing, and I will need to repeat the 'execute then correct' workflow many time. Which makes debugging painful.

Generally we use SoftAssertions http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#soft-assertions `SoftAssertions.assertSoftly(softly -> {...});`